packages/core/src/utils/security/index.ts

```diff
@@ -0,0 +1,49 @@
+import { fold, isLeft, isRight, Left, left, mapLeft } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get, identity } from 'lodash';
+import { IPrismDiagnostic } from '../../types';
+import { securitySchemeHandlers } from './handlers';
+import { SecurityScheme } from './handlers/types';
+
+function getAllInvalidSec(invalidSecuritySchemes: Array<Left<IPrismDiagnostic>>): IPrismDiagnostic {
+  const firstLeftValue = pipe(
+    invalidSecuritySchemes[0],
+    fold<IPrismDiagnostic, unknown, IPrismDiagnostic>(identity, identity),
+  );
+
+  if (firstLeftValue.code !== 401 || invalidSecuritySchemes.length === 1) {
+    return firstLeftValue;
+  } else {
+    const allWWWAuthHeaders = invalidSecuritySchemes.reduce((accumulator, currentValue) => {
+      return pipe(
+        currentValue,
+        mapLeft(err => [accumulator, get(err, ['headers', 'WWW-Authenticate'])].filter(identity).join(', ')),
+        fold(authHeader => authHeader || '', () => ''),
+      );
+    }, '');
+
+    if (allWWWAuthHeaders !== '')
+      firstLeftValue.tags ? firstLeftValue.tags.push(allWWWAuthHeaders) : (firstLeftValue.tags = [allWWWAuthHeaders]);
```

Comment:
`if` statement without `{}` is just a shoot in the foot. Also, why `push`? Let's construct it without mutations.

Reply:
There might be other value in the array, and I didn't want to cancel these.